modules/api/src/main/java/org/apache/ignite/sql/Session.java

```diff
@@ -68,11 +69,57 @@ default ResultSet execute(@Nullable Transaction transaction, String query, @Null
      * @param arguments Arguments for the statement.
      * @return SQL query results set.
      */
-    default ResultSet execute(@Nullable Transaction transaction, Statement statement, @Nullable Object... arguments) {
+    default ResultSet<SqlRow> execute(@Nullable Transaction transaction, Statement statement, @Nullable Object... arguments) {
         Objects.requireNonNull(statement);
 
         try {
-            return new SyncResultSetAdapter(executeAsync(transaction, statement, arguments).join());
+            return new SyncResultSetAdapter<>(executeAsync(transaction, statement, arguments).join());
+        } catch (CompletionException e) {
+            throw IgniteException.wrap(e);
+        }
+    }
+
+    /**
+     * Executes single SQL statement and maps results to objects with the provided mapper.
+     *
+     * @param transaction Transaction to execute the statement within or {@code null}.
+     * @param mapper Mapper.
+     * @param query SQL query template.
+     * @param arguments Arguments for the statement.
+     * @return SQL query results set.
+     */
+    default <T> ResultSet<T> execute(
```

Comment:
and again, let's add description for type parameter here and every where in public API.

Reply:
Done